GSoC2022.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Google Summer of Code Project Ideas 2022
+layout: survey
+---
+
+# JabRef in Google Summer of Code 2022
+
+This page lists a number of ideas for potential projects to be carried out by the students participating in Google Summer of Code 2022.
+This is by no means a closed list, so the students can feel free to propose alternative activities related to the project (the [list of feature requests](http://discourse.jabref.org/c/features) and the [GitHub issue tracker](https://github.com/JabRef/jabref/issues) might serve as an additional source of inspiration).
+Students are strongly encouraged to discuss their ideas with the developers and the community to improve their proposal until submission (e.g, using the [Gitter Channel](https://gitter.im/JabRef/jabref) or the [forum](http://discourse.jabref.org/)).
+It's also a good idea to start working on [one of the smaller issues](https://github.com/JabRef/jabref/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) to make yourself familiar with [the contribution process](https://github.com/JabRef/jabref/blob/master/CONTRIBUTING.md).
+
+## Communication with the JabRef-Word-Addin
+
+* *Description:*
+  During the last GSOC 2021 a React based [Microsoft Word-Extension](https://github.com/JabRef/JabRef-Word-Addin) has been developed. It provides basic functionality of an add-in for inserting and modifiyng citations into Microsoft-Word. For simplicity it has been initially developed to communicate with Jabref-Online, a server based application under development and not yet with the main desktop application.
+  The goal of this project is to enable the communication and synchronizing of citation data from the desktop application JabRef to the Microsoft-Word-Addin.
```

Comment:
How would such a communication look like? Start a webserver in JabRef that serves the necessary API? Probably its easier to only support JabRef > JabRef Online > Word extension, although it would mean that people have to register with JabRef Online.

Reply:
I thought it's maybe possible to use web sockets and have a graph ql server inerface in JabREf

Reply:
I guess that is technically possible, but do we really want to maintain this? Would also make it harder to evolve the API as the JabRef client installed by a user would still serve an older version.

Reply:
We removed that for now, we should discuss it nonetheless in the next devcall on Friday!

Reply:
The aim is to offer a desktop-only solution, because in certain settings a) using cloud applications or b) each additional software locally installed causes effort.

Regarding the evolution of the API: Move fast and break things. ^^ - As soon as the [Word Addin](https://github.com/JabRef/JabRef-Word-Addin) has released version 1.0.0, we should follow semantic versioning there. Then its liek in ohter software: JabRef 6.1 works only with Word-Addin 1.0..0, JabRef 6.2 with Word-Addin 1.1, ...

Reply:
This is not entirely true, it runs locally in a WebView context https://docs.microsoft.com/en-us/office/dev/add-ins/concepts/browsers-used-by-office-web-add-ins 
>Office Add-ins are web applications that are displayed using iFrames when running in Office on the web. In Office for desktop and mobile clients, Office Add-ins use an embedded browser control (also known as a webview). Add-ins also need a JavaScript engine to run the JavaScript. Both the embedded browser and the engine are supplied by a browser installed on the user's computer.

Reply:
That's true, but the website that you show in the webview still needs to be hosted somewhere. This can be done of course also using JabRef as the webserver, but then you have to worry about how to install it etc.

I'm very much in favor of having an easy integration of the word plugin. But I fear this has to go through JabRef Online. Maybe @mohit038 has some ideas for alternative ways to get it running.